### ISSUE TITLE
Manual setWasmUrl causes fetch for wasm file for every animation [@lottiefiles/dotlottie-react]

### ISSUE THREAD

**Issue by aalcott14** [LottieFiles/dotlottie-web#402]
## Overview
I'm attempting to use `DotLottieWorkerReact` and use the `setWasmUrl` method to retrieve the wasm file from my own server rather than the public CDN (for security purposes). It seems to work except that the fetch to the wasm url gets made for every single animation rather than a single fetch, like what occurs if `setWasmUrl` is not invoked and defers to the CDN.

I'm wondering what is causing the fetch to get made subsequent times when this does not occur without the use of `setWasmUrl`.

**Only a single fetch occurs when using `DotLottieReact` component instead of `DotLottieWorkerReact`**

Here's an example of my code:
```
import React, { MutableRefObject, useEffect, useState } from 'react';
import { DotLottieWorker, DotLottieWorkerReact, DotLottieWorkerReactProps, setWasmUrl } from '@lottiefiles/dotlottie-react';
import wasmURL from '@lottiefiles/dotlottie-web/dist/dotlottie-player.wasm?url';

setWasmUrl(`${window.location.origin}${wasmURL}`);

export interface LottieWebWorkerProps extends Omit<DotLottieWorkerReactProps, 'dotLottieRefCallback'> {
  lottieRef: MutableRefObject<DotLottieWorker>
  onLoad?: () => void;
  onComplete?: () => void;
}

export default function LottieWebWorker({ lottieRef, onLoad, onComplete, ...rest }: LottieWebWorkerProps) {
  const [dotLottie, setDotLottie] = useState<DotLottieWorker | null>(null);

  useEffect(() => {
    if (dotLottie) {
      if (lottieRef) {
        lottieRef.current = dotLottie;
      }
      if (onLoad) {
        dotLottie.addEventListener('load', onLoad);
      }
      if (onComplete) {
        dotLottie.addEventListener('complete', onComplete);
      }
    }

    return () => {
      if (dotLottie) {
        if (onLoad) {
          dotLottie.removeEventListener('load', onLoad);
        }
        if (onComplete) {
          dotLottie.removeEventListener('complete', onComplete);
        }
      }
    }
  }, [dotLottie, onLoad, onComplete]);

  return (
    <DotLottieWorkerReact
      dotLottieRefCallback={setDotLottie}
      {...rest}
    />
  );
}
```

...

## Labels

- [ Type: Bug] Add the `Type: Bug` label to this issue.

**Comment by theashraf** [comment]
@aalcott14 fetching the WASM module should occur only once, even within a worker, as you mentioned, since all dotLottie instances created by workers share the same memory.

The only exception is if you're using the `workerId` prop, which allows you to group animations into separate workers. Is that the case ?

**Comment by theashraf** [comment]
reproducible on the [perf-test page](https://lottiefiles.github.io/dotlottie-web/perf-test?seed=aG9sYSxmbHlfaW5fYmVha2VyLGhvbGRhbmltYXRpb24sYWxpZW4sc3RhcnN0cmlwcyxmaWVyeV9za3VsbCxsYV9jb21tdW5hdXRlLGVycm9yXzQwNCxmaWVyeV9za3VsbCxhbnViaXMsY29uZmV0dGkscHJvcGVydHlfbWFya2V0LGZvb2RyYXRpbmcsaG91cmdsYXNzLGZsaXBwaW5nX3BhZ2UsZWFydGhfaG91cixjb25mZXR0aUJpcmQsZGFuY2luZ19zdGFyLGNhcmRfaG92ZXIseWFybl9sb2FkaW5nLHBvbHlzdGFyLHRocmVhZHMsZmxpcHBpbmdfcGFnZSxjb25mZXR0aTIsbG9hZGluZ19yZWN0YW5nbGUsYW51YmlzLHJ1Zm8saXNvbWV0cmljLHNlYXdhbGssb3Blbl9lbnZlbG9wZSxkb3duLGZvb2RyYXRpbmcscmFkYXIsaG9sYSxob2xkYW5pbWF0aW9uLHNlYXdhbGssbWF0ZXJpYWxfd2F2ZV9sb2FkaW5nLGFudWJpcyxncmFkaWVudF9iYWNrZ3JvdW5kLGxhX2NvbW11bmF1dGUsdWtfZmxhZyxza3VsbGJveSx0ZXh0LGZpZXJ5X3NrdWxsLGZpZXJ5X3NrdWxsLGxhX2NvbW11bmF1dGUscnVmbyxmb29kcmF0aW5nLGhvbGEsZm9vZHJhdGluZyx5YXJuX2xvYWRpbmcsZmllcnlfc2t1bGwsYV9tb3VudGFpbixmdW5reV9jaGlja2VuLGNvbmZldHRpQmlyZCxkb3duLGZpZXJ5X3NrdWxsLGxhX2NvbW11bmF1dGUsc2FtcGxlLGdyYWRpZW50X3NsZWVweV9sb2FkZXIsbG9sb193YWxrLGZ1bmt5X2NoaWNrZW4sc3RhcmJ1cnN0LGNvaW4sNTM0NC1ob25leS1zYWNrLWh1ZCxtZXJnaW5nX3NoYXBlcyxkb2RlY2FoZWRyb24sZGFuY2luZ19zdGFyLG1lc3NhZ2UsY29pbixjb2luLHRleHQyLGZsaXBwaW5nX3BhZ2UsY29uZmV0dGkyLGZ1bl9hbmltYXRpb24sY2FsY3VsYXRvcix0ZXh0LHNrdWxsYm95LHNrdWxsYm95LHRleHRfYW5pbSxyaXBwbGVfbG9hZGluZ19hbmltYXRpb24sYmF0bWFuLG1vcnBoaW5nX2FuaW0sZ2VvbWV0cmljLHNhZF9lbW9qaSwxNjQzLWV4cGxvZGluZy1zdGFyLGdsb3dfbG9hZGluZyxzd2luZ2luZyxkb3duLHJhZGFyLGdhbWVfZmluaXNoZWQsZWFydGhfaG91cixzYWRfZW1vamksZmllcnlfc2t1bGwsNTMxNy1maXJld29ya2RzLG9wZW5fZW52ZWxvcGUsdHJhaW4sZGFuY2luZ19zdGFyLDI3NzQ2LWpveXBpeGVscy1wYXJ0eWluZy1mYWNlLWVtb2ppLWFuaW1hdGlvbiwxNjQzLWV4cGxvZGluZy1zdGFy&player=dotlottie-web/worker&count=100)


<img width="500" alt="image" src="https://github.com/user-attachments/assets/2dc10099-42e2-4849-a1d9-8dd5e01cc93d">

**Comment by aalcott14** [comment]
Hi @theashraf, thanks for taking a look! Yes, as you can see on the perf test page, this issue becomes quite problematic from a performance perspective as the requests pile up

**Comment by theashraf** [comment]
@aalcott14 fixed in @lottiefiles/dotlottie-react@0.10.1
